### ISSUE TITLE
[DOXIA-737] [REGRESSION] Making EmitAnchorsForIndexableEntries by default does not extract document titles anymore

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia#864]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[DOXIA-737](https://issues.apache.org/jira/browse/DOXIA-737?redirect=false)** and commented

While working on https://github.com/apache/maven-site/pull/525 the following happens:

Title before:
{project} -- {document}

Title now:

{project}

The document title is not extracted from the document anymore.
If I do the following:

```
diff --git a/doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java b/doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java
index 544b6b9..b6a7d1d 100644
--- a/doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java
+++ b/doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java
@@ -330,7 +330,6 @@ public class DefaultSiteRenderer implements Renderer {
             if (!isConfigured) {
                 // DOXIASITETOOLS-146 don't render comments from source markup
                 parser.setEmitComments(false);
-                parser.setEmitAnchorsForIndexableEntries(true);
             }

             // TODO: DOXIA-111: the filter used here must be checked generally.
 
```

Everything works again. I bet that the `SiteRendererSink` does not work well with the new approach in Doxia.

---

**Affects:** 2.0.0-M11

**Remote Links:**
- [GitHub Pull Request #214
](https://github.com/apache/maven-doxia/pull/214)

**Backported to:** [2.0.0-M12](https://github.com/apache/maven-doxia/milestone/44?closed=1)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin), can you investigate because you know the new Doxia code best? Maybe the actual bug is in Doxia, not here.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Found it, truly a regression. `IndexingSink` was not passing the title text to the super class.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [c35e61cf6119f67b992c80aa9a971a1eb6975a8d](https://gitbox.apache.org/repos/asf?p=maven-doxia.git;a=commit;h=c35e61cf6119f67b992c80aa9a971a1eb6975a8d).
